### ISSUE TITLE
About implementation of the generation method of lambdaKG

### ISSUE THREAD

**Issue by yusuke1997** [zjunlp/PromptKG#38]
你好！
I have a question about KGT5 and GenKGC for KGC tasks- are there any implementation differences between the two?
I seem to be very similar, maybe the difference is using t5-small or bart-base for the pre-train model.
Data process is the same, lit_models/transformers.py are nearly the same, and models are slightly different but I could hardly find a difference.
Also, it seems that the data pre-processing of each paper is different. For example, KGT5 uses "predict_tail: ", "predict_head", and "predict_answer" as prompts, and GenKGC uses type information, and the insertion position of the descriptions is at the beginning of the input.
Other than that, I think there are some differences between the lambdaKG implementation and both methods described in the paper.
But the lambdaKG paper only says that generation-based models are given the tag [inverse].
I would like to ask whether the lambdaKG implementation does not distinguish between minor differences, or whether it is an implementation mistake.
I would like to know if you have also compared the differences in accuracy in such cases.

best,

**Comment by zxlzr** [comment]
Hello, thanks for being interested in our work. 

Sorry, LambdaKG is still under development. We implement slightly different input formats for the KGT5 and GenKGC models (we will address this in the paper).

To use the same data processor and training modules, we use the basic T5 model for KGT5, which requires slightly different input from the original T5 paper. So the different input format may affect the model’s performance. 

Specifically, 

- We use the input with [inverse] to predict the head entity.
- We set KGT5 with origin LM T5-base with only KG training (for  comparison).
- We only use KGT5 for KG link prediction task.

In the original GenKGC repository (https://github.com/zjunlp/PromptKG/tree/main/research/GenKGC), triplets were used as prompts, and prefix tree constraint encoding was employed. However, since LambdaKG is still undergoing rapid iteration, we have implemented a simpler version of GenKGC without prompts. 

We plan to migrate the original GenKGC code to LambdaKG soon.

By the way, this tool is designed for easy use of PLM based on KGE, if you want to reproduce the results, it is recommended to use the original repository  from the paper.

Best.

**Comment by yusuke1997** [comment]
Thank you for your very quick reply!! I understand clearly.
I think lambdaKG is the best tool for PLM-based KGE. I hope it will be completed soon.
I would use the original repository for GenKGC.
Another question: is there a roadmap for development for lambdaKG?
My concern is that development will stop. The modifications have stopped for a couple of months.
I would like to know how long it is expected to take to migrate GenKGC or the other models, features.

Best regards,

**Comment by zxlzr** [comment]
Hi, we plan to release an updated version in August. Progress has slowed down recently due to the graduation of the main developer.

**Comment by yusuke1997** [comment]
Thank you for this very important information!
lambdaKG is a good library and I look forward to future updates.
Good luck!!

**Comment by zxlzr** [comment]
Hi, we have updated the GenKGC and will continue fix new bugs soon.
